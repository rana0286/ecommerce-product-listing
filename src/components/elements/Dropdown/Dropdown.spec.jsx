import React from 'react';
import { mount, shallow } from 'enzyme';
import DropDown from './Dropdown';
import renderer from 'react-test-renderer';

describe('src/components/elements/Dropdown', () => {
  const options = [
    {
      label: 'Filter by size',
      value: '',
    },
    {
      label: 'XS',
      value: 'XS',
    },
    {
      label: 'S',
      value: 'S',
    },
    {
      label: 'M',
      value: 'M',
    },
    {
      label: 'L',
      value: 'L',
    },
    {
      label: 'XL',
      value: 'XL',
    },
  ];

  const onChangeMock = jest.fn();
  it('Drop down component should render', () => {
    const component = shallow(<DropDown options={options} onSelect={onChangeMock}/>);
    const tree = renderer.
      create(<DropDown options={options} onSelect={onChangeMock}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(component).toBeTruthy();
  });

  it('Drop down component should call onSelect with selected value', () => {
    const event = {
      preventDefault() {},
      target: { value: 'test-value' }
    };
    const component = mount(<DropDown options={options} onSelect={onChangeMock}/>);
    component.find('select').simulate('change', event);
    expect(onChangeMock).toBeCalledWith('test-value');
  });
});
