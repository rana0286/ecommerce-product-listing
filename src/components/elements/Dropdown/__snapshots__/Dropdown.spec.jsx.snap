// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Dropdown Drop down component should render 1`] = `
<div
  className="select-box "
>
  <label
    className="select-box__label"
    htmlFor="select-box"
  >
    
  </label>
  <select
    className="select-box__select"
    id="select-box"
    onChange={[Function]}
    onSelect={[MockFunction]}
    selected=""
    value=""
  >
    <option
      value=""
    >
      Filter by size
    </option>
    <option
      value="XS"
    >
      XS
    </option>
    <option
      value="S"
    >
      S
    </option>
    <option
      value="M"
    >
      M
    </option>
    <option
      value="L"
    >
      L
    </option>
    <option
      value="XL"
    >
      XL
    </option>
  </select>
</div>
`;
